### PR TITLE
Add !hoogle

### PR DESCRIPTION
**Pull request by MaxwellBo**
*With newlines for `--verbose`*
withnewlines

*Without newlines for `--verbose`*
withoutnewlines

### CODE REVIEW

File: uqcsbot/scripts/hoogle.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    return "https://www.haskell.org/hoogle/?mode=json&hoogle=" + unescaped + "&start=0&count=10"
+
+def pretty_hoogle_result(result: dict, is_verbose: bool) -> str:
+    url = result['location']
+    type_sig = result['self']
+    docs = result['docs']
+
+    if is_verbose:
+        return f"`{type_sig}` <{url}|link> {docs}"
```
Comment (TRManderson) [inline uqcsbot/scripts/hoogle.py]: Formatting  could definitely be better for verbose mode

Comment (MaxwellBo) [reply uqcsbot/scripts/hoogle.py]: `\n` between the link and the docs?

File: uqcsbot/scripts/hoogle.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    command_args = command.arg.split() if command.has_arg() else []
+
+    verbose = False
+
+    if '--verbose' in command_args:
+        command_args.remove('--verbose')
+        verbose = True
+
+    if len(command_args) == 0:
+        bot.post_message(command.channel, handle_hoogle.__doc__)
```
Comment (mitchmcdee) [inline uqcsbot/scripts/hoogle.py]: I like this, good use of `__doc__` 👍 i reckon we'll adopt something similar for other scripts
maybe prepend with `usage: ` just to be 100% clear?

File: uqcsbot/scripts/hoogle.py (hunk 1 above)
Comment (mitchmcdee) [reply uqcsbot/scripts/hoogle.py]: throw up a screenshot of both options?

Comment (mitchmcdee) [reply uqcsbot/scripts/hoogle.py]: (so we can see + help decide)

File: uqcsbot/scripts/hoogle.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+def handle_hoogle(command: Command):
+    '''
+    `!hoogle [--verbose] <TYPE_SIGNATURE>` - Queries the Hoogle Haskell API search engine, 
+    searching Haskell libraries by either function name, or by approximate type signature.
+    '''
+    command_args = command.arg.split() if command.has_arg() else []
+
+    verbose = False
+
+    if '--verbose' in command_args:
```
Comment (mitchmcdee) [inline uqcsbot/scripts/hoogle.py]: might as well add `-v` too

Comment (TRManderson) [reply uqcsbot/scripts/hoogle.py]: may as well just use argparse at this point 😂

Comment (mitchmcdee) [reply uqcsbot/scripts/hoogle.py]: yis.

Comment (MaxwellBo) [reply uqcsbot/scripts/hoogle.py]: isn't `-v` always for version?

Comment (TRManderson) [reply uqcsbot/scripts/hoogle.py]: curl, tar, mv, cp

File: uqcsbot/scripts/hoogle.py (hunk 1 above)
Comment (MaxwellBo) [reply uqcsbot/scripts/hoogle.py]: Added screenshots

File: uqcsbot/scripts/hoogle.py (hunk 3 above)
Comment (mitchmcdee) [reply uqcsbot/scripts/hoogle.py]: I thought `-V` was version i.e. python

File: uqcsbot/scripts/hoogle.py (hunk 1 above)
Comment (mitchmcdee) [reply uqcsbot/scripts/hoogle.py]: `\n` lgtm

File: uqcsbot/scripts/hoogle.py (hunk 3 above)
Comment (MaxwellBo) [reply uqcsbot/scripts/hoogle.py]: ah fair enough, I'll add `-v` then

**Pull request merged**